src/Symfony/Component/HttpKernel/HttpKernel.php

```diff
@@ -82,6 +87,18 @@ public function handle(Request $request, int $type = HttpKernelInterface::MAIN_R
                 throw $e;
             }
 
+            return $this->handleThrowable($e, $request, $type);
```

Comment:
Wouldn't it reduce the diff while keeping only one `catch` block to do something like this:
```php
try {
    // ...
} catch (\Throwable $e) {
    if (!$e instanceof \Exception && !$this->catchThrowable) {
        throw $e;
    }
    if ($e instanceof RequestExceptionInterface) {
        $e = new BadRequestHttpException($e->getMessage(), $e);
    }
    if (false === $catch) {
        $this->finishRequest($request, $type);

        throw $e;
    }

    return $this->handleThrowable($e, $request, $type);
}
```
?

Reply:
Not by a hole not, no. I don’t think so. (I’ll try). 

I made two catch blocks to explicitly show this new code path. (And to be 100% sure I (and anybody after me) don’t make mistakes with the exception code path)

Reply:
I added a comment in my example as I am talking about both the current diff and the one we'll get with 7.0.

Reply:
Thank you.